backend/functions/src/index.ts

```diff
@@ -111,27 +110,73 @@ export const itemitemCollaborativeFiltering = functions.https.onRequest(
       }
     }
 
-    //Generating user recommendations
-    const recommendations: number[] = [];
+    // Generating recommendations for Item Based Collaborative Filtering
+    const recommendationsI: number[] = [];
     for (let i = 0; i < problemSet.length; i++) {
       let prsum: number = 0;
       let sum: number = 0;
       for (let j = 0; j < problemSet.length; j++) {
-        prsum += dataset[userList.indexOf(user)][j] * iimat[i][j];
+        prsum += dataset[userIndex][j] * iimat[i][j];
         sum += iimat[i][j];
       }
       if (sum == 0) {
-        recommendations[i] = 0;
+        recommendationsI[i] = 0;
       } else {
-        recommendations[i] = prsum / sum;
+        recommendationsI[i] = prsum / sum;
+      }
+    }
+
+    //#################################User Based Collaborative Filtering#################################
+    //Generating the user-similarity matrix.
+    const uumat: number[][] = [];
+    for (let i = 0; i < problemSet.length; i++) {
```

Comment:
`problemSet` should be replaced by `userList`  ?

Reply:
Yes, I will resolve it in the next PR.